src/MediaForm.php

```diff
@@ -94,12 +94,12 @@ public function save(array $form, FormStateInterface $form_state) {
   /**
    * {@inheritdoc}
    */
-  public function validate(array $form, FormStateInterface $form_state) {
-    parent::validate($form, $form_state);
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    parent::validateForm(&$form, $form_state);
     /** @var \Drupal\media_entity\MediaInterface $entity */
     $entity = $this->buildEntity($form, $form_state);
     /** @var \Drupal\media_entity\MediaInterface $entity_unchanged */
-    if ($entity->id()) {    
+    if ($entity->id()) {
```

Comment:
In long-term, this method should go away completely.

The changed stuff should now work in a generic way using entity validation. If your entity type implements EntityChangedInterface, then you automatically get the EntityChanged constraint and that should do that validation for you.

So I'm pretty sure you can just drop that part if you do that.

And the second part below about the type validation should also be a constraint, so that it also works when creating things through REST and so on. But that can be done later :)